include/simdjson/implementation.h

```diff
@@ -83,6 +101,18 @@ class implementation {
    * @return the error code, or SUCCESS if there was no error.
    */
   WARN_UNUSED virtual error_code minify(const uint8_t *buf, size_t len, uint8_t *dst, size_t &dst_len) const noexcept = 0;
+  
+  
+  /**   
+   * Validate the UTF-8 string.
+   *
+   * Overridden by each implementation.
+   *
+   * @param buf the string to validate.
+   * @param len the length of the string in bytes.
+   * @return true if and only if the string is valid UTF-8.
```

Comment:
Do we need to mention the fact that we're not doing anything special with surrogate pairs? I think maybe we don't need to, but I'm not sure what its status is re: UTF-8.

Reply:
> Do we need to mention the fact that we're not doing anything special with surrogate pairs?

Surrogates in UTF-8 are invalid. We are just straight-out applying the specification here. If someone wants to use UTF-16 surrogates in UTF-8, God help them...